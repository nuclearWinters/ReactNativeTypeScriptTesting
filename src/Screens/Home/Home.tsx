import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  TouchableOpacity,
  StatusBar,
} from 'react-native';
import {Header, Colors} from 'react-native/Libraries/NewAppScreen';
import {StackScreenProps} from '@react-navigation/stack';
import {RootStackParamList} from '../../../App';

export type Props = StackScreenProps<RootStackParamList, 'Home'>;

declare const global: {HermesInternal: null | {}};

const Home: React.FC<Props> = (props) => {
  const {navigation} = props;
  return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView>
        <ScrollView
          contentInsetAdjustmentBehavior="automatic"
          style={styles.scrollView}>
          <Header />
          {global.HermesInternal == null ? null : (
            <View style={styles.engine}>
              <Text style={styles.footer}>Engine: Hermes</Text>
            </View>
          )}
          <TouchableOpacity
            style={styles.sectionContainer}
            onPress={() => {
              navigation.navigate('Counter', {
                itemId: 10,
              });
            }}>
            <Text style={styles.sectionTitle}>
              Ir a la pantalla de Contador
            </Text>
            <Text style={styles.sectionDescription}></Text>
          </TouchableOpacity>
        </ScrollView>
      </SafeAreaView>
    </>
  );
};

const styles = StyleSheet.create({
  scrollView: {
    backgroundColor: Colors.lighter,
  },
  engine: {
    position: 'absolute',
    right: 0,
  },
  body: {
    backgroundColor: Colors.white,
  },
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.dark,
  },
  highlight: {
    fontWeight: '700',
  },
  footer: {
    color: Colors.dark,
    fontSize: 12,
    fontWeight: '600',
    padding: 4,
    paddingRight: 12,
    textAlign: 'right',
  },
});

export default Home;
