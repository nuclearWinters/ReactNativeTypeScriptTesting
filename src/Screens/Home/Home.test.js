import React from 'react';
import {render} from '@testing-library/react-native';
import Home from './Home';

test('Home', () => {
  const {getByTestId, getByText, queryByTestId, baseElement} = render(
    <Home props={{navigation: {navigate: () => {}}}} />,
  );
  const button = getByText('Ir a la pantalla de Contador');
  expect(button).toBeTruthy();
  const title = getByText('Welcome to React');
  expect(title).toBeTruthy();
  expect(baseElement).toMatchSnapshot();
});
