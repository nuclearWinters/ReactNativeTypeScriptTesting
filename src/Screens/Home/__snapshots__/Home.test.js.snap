// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView>
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={
          Object {
            "backgroundColor": "#F3F3F3",
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "#F3F3F3",
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            }
          }
        >
          <Image
            accessibilityRole="image"
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "marginBottom": -192,
                  "marginLeft": -128,
                  "opacity": 0.2,
                  "overflow": "visible",
                  "resizeMode": "cover",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 40,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            Welcome to React
          </Text>
        </View>
        <TouchableOpacity
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "fontWeight": "600",
              }
            }
          >
            Ir a la pantalla de Contador
          </Text>
          <Text
            style={
              Object {
                "color": "#444",
                "fontSize": 18,
                "fontWeight": "400",
                "marginTop": 8,
              }
            }
          />
        </TouchableOpacity>
      </ScrollView>
    </SafeAreaView>
  </View>
</View>
`;
